app/src/main/java/com/truongdc21/mediatree/ui/activity/RegisterActivity.kt

```diff
@@ -0,0 +1,63 @@
+package com.truongdc21.mediatree.ui.activity
+
+import androidx.activity.viewModels
+import androidx.lifecycle.Observer
+import com.truongdc21.mediatree.MainActivity
+import com.truongdc21.mediatree.R
+import com.truongdc21.mediatree.base.BaseActivity
+import com.truongdc21.mediatree.databinding.ActivityRegisterBinding
+import com.truongdc21.mediatree.utils.showToast
+import com.truongdc21.mediatree.utils.switchActivity
+import com.truongdc21.mediatree.viewmodel.AuthViewModel
+import dagger.hilt.android.AndroidEntryPoint
+
+@AndroidEntryPoint
+class RegisterActivity : BaseActivity<ActivityRegisterBinding>(ActivityRegisterBinding::inflate) {
+
+    private val mViewModel : AuthViewModel by viewModels()
+
+    override fun initViews() {
+        binding.apply {
+            imgBack.setOnClickListener {
+                finish()
+            }
+            btnRegister.setOnClickListener {
+                registerFirebase()
+            }
+        }
+    }
+
+    override fun initData() {
+        checkRegister()
+        showToastError()
+    }
+
+    private fun registerFirebase() {
+        binding.apply {
+            val srtEmail = edtEmail.text.toString().trim()
+            val srtPass = edtPassword.text.toString().trim()
+            val srtComfirmPass = edtComfirmPassword.text.toString().trim()
+            if (srtEmail.isEmpty()  ||  srtPass.isEmpty() || srtComfirmPass.isEmpty()){
+                return
+            }
+            if (srtPass != srtComfirmPass){
+                this@RegisterActivity.showToast(resources.getString(R.string.password_incorrect))
+                return
+            }
+            mViewModel.signUpAuth(srtEmail, srtPass)
+        }
+    }
+
+    private fun checkRegister(){
+        mViewModel.signUpAuthLiveData.observe(this , Observer { isCheck ->
+            this.showToast(resources.getString(R.string.register_success))
+            finish()
+        })
+    }
+
+    private fun showToastError(){
+        mViewModel.errorExceptionFirebase.observe(this , Observer { message ->
```

Comment:
```suggestion
        mViewModel.errorExceptionFirebase.observe(this , { message ->
```
Bỏ đi đc ko , nó là lambda mà